### ISSUE TITLE
Research models and architectures

### ISSUE THREAD

**Issue by natborn2** [natborn2/birdsai#3]

**Comment by natborn2** [comment]
Models and architectures to consider:
- YOLO, one of the most popular architectures for CV tasks
    - [Ultralytics](https://docs.ultralytics.com/) [YOLOv8](https://github.com/ultralytics/ultralytics), the latest version of the acclaimed real-time object detection and image segmentation model
    - read [Straker et al. 2023](https://www.sciencedirect.com/science/article/pii/S2667393223000169#cebib0010): YOLO implementation for ITC segmentation
    - see also [Tutorial for YOLO in FiftyOne](https://docs.voxel51.com/tutorials/yolov8.html)
    - for reference [Redmon et al. 2016](https://www.cv-foundation.org/openaccess/content_cvpr_2016/html/Redmon_You_Only_Look_CVPR_2016_paper.html): The original YOLO paper 
- U-Net
    - [Hu & Jung 2022](https://isprs-archives.copernicus.org/articles/XLIII-B3-2021/61/2021/isprs-archives-XLIII-B3-2021-61-2021.pdf): Implementation of U-Net for ITC delineation 
    - [Freudenberg et al. 2022](https://link.springer.com/article/10.1007/s00521-022-07640-4) Another implementation of U-Net for ITC delineation 
- Faster R-CNN
    - [Beloiu et al. 2023](https://www.mdpi.com/2072-4292/15/5/1463): Implementation for ITC delineation
- DeepForest
    - [Weinstein et al. 2019](https://www.mdpi.com/2072-4292/11/11/1309) DeepForest Deep Learning RGB Detection uses a RetinaNet one-stage detector with a pretrained ResNet-50 backbone

Model comparisons and reviews: 
- Backbones-Review: Feature Extraction Networks for Deep Learning and Deep Reinforcement Learning Approaches: [Elharroussa et al. 2022](https://arxiv.org/pdf/2206.08016.pdf)
- A Systematic Review of Individual Tree Crown Detection and Delineation with Convolutional Neural Networks (CNN): [Zhao et al. 2023](https://link.springer.com/article/10.1007/s40725-023-00184-3) - Must read this

- A Review of Researches on Deep Learning in Remote Sensing Application [Zhu et al. 2019](https://www.scirp.org/journal/paperinformation?paperid=89791)

Read and make notes on these references. Put notes in Mendeley. Digest notes here.

**Comment by natborn2** [comment]
Straker+23:
- Review methods for ITC in UAV Lidar / ALS point cloud data
- The YOLOv5 code repo by Jocher+22 is accessible
- YOLO outperforms Mask-CNN and R-CNN
- Good description of practical application of generic ML model to ITC problem: train/val/test split; 2-step training procedure to maximise small training sample; consideration of overfitting; IoU calculation to classify TP/TN/FP/FN; performance metrics.

**Comment by natborn2** [comment]
Hu + Jung 2022:
- First implementation of U-Net, Attention U-Net, Residual U-Net in ITC segmentation in aerial RGB optical imagery
- Refer to Zhu+19 for a general review of Deep Learning in Remote Sensing applications
- Use of a weight map to increase the weights of boundary pixels in the loss function: a gaussian function of the sum of the distance to the borders of the nearest and second nearest crowns
- Fairly brief description of the method, 
- High accuracy in trained models - can probably be attributed to the high quality of the training and test data, and the fact that tree crowns in these images are relatively distinct
- Note that the weight map was very beneficial in improving delineation accuracy. However, results are sensitive to the choice of the parameters for this weight map, and these parameters are site-specific

**Comment by natborn2** [comment]
Freudenberg+ 22 in Mendeley: https://www.mendeley.com/reference-manager/reader-v2/817d1403-4b69-3d42-8f74-cf49699351fb/1661f175-f3b8-b619-49a0-b949296801e0
- deep learning-based framework for individual tree crown delineation in aerial and satellite images
- in addition to ITC delineation, the method provides a tree cover mask for areas that are not separable.
- used 30cm Worldview satellite imagery in an urban area and 5cm aerial imagery in a densely forested area
- trained with low amount of training data
- source code and pretrained weights are publicly available at https://github.com/AWF-GAUG/TreeCrownDelineation
-  ...
